src/integrations/diesel.rs

```diff
@@ -99,23 +108,25 @@ fn run_barrel_migration(migration: &Path) -> (String, String) {
     let dir = TempDir::new("barrel").unwrap();
     fs::create_dir_all(&dir.path().join("src")).unwrap();
 
-    /* Add a Cargo.toml file */
-    let ct = dir.path().join("Cargo.toml");
-    let mut cargo_toml = File::create(&ct).unwrap();
-    cargo_toml
-        .write_all(
-            b"# This file is auto generated by barrel
+    let (feat, ident) = get_backend_pair();
+    let mut toml: String = "# This file is auto generated by barrel
 [package]
 name = \"tmp-generator\"
 description = \"Doing nasty things with cargo\"
 version = \"0.0.0\"
 authors = [\"Katharina Fey <kookie@spacekookie.de>\"]
-
 # TODO: Use same `barrel` dependency as crate
-[dependencies]
-barrel = { git = \"https://github.com/spacekookie/barrel\", features = [\"pg\"]  }",
-        )
-        .unwrap();
+[dependencies]"
+        .into();
+
+    toml.push_str("barrel = { version = \"*\", features = [ ");
+    toml.push_str(&format!("{}", feat)); // Not sure why format strings are being weird here
+    toml.push_str(" ] }");
```

Comment:
The full `toml` could be generated using format! and avoid the multiple  `push_str` calls and make it easier to read.

Reply:
The problem I ran into was that the `format!` string got confused with the `{` in ways I didn't really know how to escape.

You're right, it would be much nicer to read in one block. I might take another stab at this. But it wasn't super trivial unfortunately :confused: